docker/Dockerfile

```diff
@@ -3,38 +3,226 @@
+ARG WEBRTC_REPO=https://github.com/open-webrtc-toolkit/owt-deps-webrtc.git
+ARG SVT_VER=v1.3.0
+ARG SVT_REPO=https://github.com/intel/SVT-HEVC.git
+ARG SERVER_PATH=/home/owt-server
+ARG OWT_SDK_REPO=https://github.com/open-webrtc-toolkit/owt-client-javascript.git
+ARG OWT_BRANCH=master
+
+RUN apt-get update && apt-get install -y -q --no-install-recommends python libglib2.0-dev rabbitmq-server mongodb libboost-thread-dev libboost-system-dev liblog4cxx-dev
+
+# 1. Clone OWT server source code 
+# 2. Clone licode source code and patch
+# 3. Clone webrtc source code and patch
+RUN git config --global user.email "you@example.com" && \
+    git config --global user.name "Your Name" && \
+    git clone --depth=1 -b ${OWT_BRANCH} ${OWTSERVER_REPO} && \
+
+    # Install node modules for owt
+    npm install -g --loglevel error node-gyp grunt-cli underscore jsdoc && \
+    cd owt-server && npm install nan && \
+
+    # Get openh264 for owt
+    cd third_party && \
+    mkdir openh264 && cd openh264 && \
+    wget ${OPENH264_SOURCE} && \
+    wget ${OPENH264_BINARY} && \
+    tar xzf ${OPENH264_SOURCENAME} openh264-${OPENH264_MAJOR}.${OPENH264_MINOR}.0/codec/api && \
+    ln -s -v openh264-${OPENH264_MAJOR}.${OPENH264_MINOR}.0/codec codec && \
+    bzip2 -d ${OPENH264_BINARYNAME}.bz2 && \
+    ln -s -v ${OPENH264_BINARYNAME} libopenh264.so.${OPENH264_SOVER} && \
+    ln -s -v libopenh264.so.${OPENH264_SOVER} libopenh264.so && \
+    echo 'const char* stub() {return "this is a stub lib";}' > pseudo-openh264.cpp && \
+    gcc pseudo-openh264.cpp -fPIC -shared -o pseudo-openh264.so 
+
+# Get licode for owt
+RUN cd ${SERVER_PATH}/third_party && git clone ${LICODE_REPO} && \
+    cd licode && \
+    git reset --hard ${LICODE_COMMIT} && \
+    git am ${SERVER_PATH}/scripts/patches/licode/*.patch
+
+# Install webrtc for owt
+RUN cd ${SERVER_PATH}/third_party && mkdir webrtc  && cd webrtc &&\
+    export GIT_SSL_NO_VERIFY=1 && \
+    git clone -b 59-server ${WEBRTC_REPO} src && \
+    ./src/tools-woogeen/install.sh && \
+    ./src/tools-woogeen/build.sh 
+
+# Install webrtc79 for owt
+RUN mkdir ${SERVER_PATH}/third_party/webrtc-m79 && \
+    cd ${SERVER_PATH}/third_party/webrtc-m79 && \
+    /bin/bash ${SERVER_PATH}/scripts/installWebrtc.sh
+
+# Get js client sdk for owt
+RUN cd /home && git clone --depth=1 -b ${OWT_BRANCH} ${OWT_SDK_REPO} && cd owt-client-javascript/scripts && npm install && grunt  && \
+    mkdir ${SERVER_PATH}/third_party/quic-lib && \
+    export LD_LIBRARY_PATH=/usr/local/lib/x86_64-linux-gnu && \
+    cd ${SERVER_PATH}/third_party/quic-lib && wget https://github.com/open-webrtc-toolkit/owt-deps-quic/releases/download/v0.1/dist.tgz && tar xzf dist.tgz && \
+    #Build and pack owt
+    cd ${SERVER_PATH} && export PKG_CONFIG_PATH=/usr/local/lib/x86_64-linux-gnu/pkgconfig && ./scripts/build.js -t mcu-all -r -c && \
+    ./scripts/pack.js -t all --install-module --no-pseudo --app-path /home/owt-client-javascript/dist/samples/conference
```

Comment:
If the user is installing their own app, pack.js shouldn't be called until their app has been copied to the image right? Otherwise they will just be running pack.js twice.  It will still work fine like this, but it will cause it to take longer to build if you are not using the owt-client-javascript sample app.

Reply:
We provide this Dockerfile to developers so that they can easily setup OWT building environment and then launch OWT service to figure out what they can do with OWT. User can specify the APP_PATH with "docker build --target owt-build --build-arg APP_PATH=xxx -t owt:build . ", and i will add APP_PATH arg in the Dockerfile and set the default value to owt-client-javascript path

Reply:
We provide this Dockerfile as a reference. We hope that developers can use this Dockerfile to  easily setup a OWT building environment and launch OWT service to figure out what OWT supports. Based on this consideration, we pack the sample app into the OWT build image so that they can quickly launch all OWT service and access to the default web page. After they get familiar with OWT, they can modify the Dockerfile and develop their own app and service.